spec.md

```diff
@@ -572,217 +572,152 @@ See [@sec:mem] for more details.
+In contrast with all other ground types, analog signals cannot appear on either side of a connection statement.
+Instead, analog signals are attached to each other with the commutative `attach`{.firrtl} statement.
+An analog signal may appear in any number of attach statements, and a legal circuit may also contain analog signals that are never attached.
+The only primitive operations that may be applied to analog signals are casts to other signal types.
 
-1. An uninferred reset driven by and/or driving only asynchronous resets will be inferred as asynchronous reset.
-1. An uninferred reset driven by and/or driving both asynchronous and synchronous resets is an error.
-1. Otherwise, the reset is inferred as synchronous (i.e. the uninferred reset is only invalidated or is driven by or drives only synchronous resets).
+When an analog signal appears as a field of an aggregate type, the aggregate cannot appear in a standard connection statement.
 
-`Reset`{.firrtl}s, whether synchronous or asynchronous, can be cast to other types.
-Casting between reset types is also legal:
+As with integer types, an analog type can represent a multi-bit signal.
+When analog signals are not given a concrete width, their widths are inferred according to a highly restrictive width inference rule, which requires that the widths of all arguments to a given attach operation be identical.
 
 ``` firrtl
-input a : UInt<1>
-output y : AsyncReset
-output z : Reset
-wire r : Reset
-connect r, a
-connect y, asAsyncReset(r)
-connect z, asUInt(y)
+Analog<1>  ; 1-bit analog type
+Analog<32> ; 32-bit analog type
 ```
 
-See [@sec:primitive-operations] for more details on casting.
+The analog type also has the inferred form: `Analog`{.firrtl} (see [@sec:width-inference]).
 
 ## Aggregate Types
 
-FIRRTL supports three aggregate types: vectors, bundles, and enumeration.
-Aggregate types are composed of ground types or other aggregate types.
+FIRRTL supports three aggregate types: vectors, bundles, and enumerations.
 
 ### Vector Types
 
 A vector type is used to express an ordered sequence of elements of a given type.
-The length of the sequence must be non-negative and known.
 
-The following example specifies a ten element vector of 16-bit unsigned integers.
+The following example specifies a 10-element vector of 16-bit unsigned integers.
 
 ``` firrtl
 UInt<16>[10]
 ```
 
-The next example specifies a ten element vector of unsigned integers of omitted but identical bit widths.
-
-``` firrtl
-UInt[10]
-```
-
-Note that any type, including other aggregate types, may be used as the element type of the vector.
-The following example specifies a twenty element vector, each of which is a ten element vector of 16-bit unsigned integers.
+Note that the element type of a vector can be any type, including another aggregate type.
+The following example specifies a 20-element vector, each of which is a 10-element vector of 16-bit unsigned integers.
 
-``` firrtl
+```firrtl
```

Comment:
Mega-nit: FWIW, the formatter is going to add a space after triple-backtick. It's fine to just handle this with a format PR.

Reply:
Wait what? Have I been writing markdown codeblocks wrong my whole life?

I'll submit a separate PR to run the formatter.